nmagent/client.go

```diff
@@ -284,6 +283,37 @@ func (c *Client) GetHomeAz(ctx context.Context) (AzResponse, error) {
 	return homeAzResponse, nil
 }
 
+// GetInterfaceIPInfo fetches the node's interface IP information from nmagent
+func (c *Client) GetInterfaceIPInfo(ctx context.Context) (Interfaces, error) {
+	req, err := c.buildRequest(ctx, &GetSecondaryIPsRequest{})
+	var out Interfaces
+
+	if err != nil {
+		return out, errors.Wrap(err, "building request")
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return out, errors.Wrap(err, "submitting request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return out, die(resp.StatusCode, resp.Header, resp.Body, req.URL.Path)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return out, die(resp.StatusCode, resp.Header, resp.Body, req.URL.Path)
+	}
+
+	err = xml.NewDecoder(resp.Body).Decode(&out)
```

Comment:
i know logging entire xml output would spam logs but we need to find a way to log if we get empty xml doc from nmagent

Reply:
@tamilmani1989 Under what conditions will that happen? We can (and should) detect that and create an error for it that says _why_ NMAgent is giving us empty xml.